sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * SyncToken class requires id, value and sequence number
+ */
+public final class SyncToken {
+    private String id, value;
+    private long sequenceNumber;
+    private String syncTokenString;
+
+    /**
+     * Create an instance of SyncToken class
+     *
+     * @param id ID of token
+     * @param value Value of Token
+     * @param sequenceNumber Token sequence number (version). Higher value means newer version of the same token.
+     */
+    private SyncToken(String id, String value, long sequenceNumber, String syncTokenString) {
+        this.id = id;
+        this.value = value;
+        this.sequenceNumber = sequenceNumber;
+        this.syncTokenString = syncTokenString;
+    }
+
+    static SyncToken fromSyncTokenString(String syncToken) {
```

Comment:
Javadocs? What are the returns we expect?
From other static method names that parse values (ie. `Integer.parseInt`), `parseSyncToken` is a better name.

Reply:
I would actually expect an InvalidArgumentException if something we expected to be a syncToken could not be parsed.

Reply:
updated